exasol/bucketfs/__init__.py

```diff
@@ -0,0 +1,148 @@
+"""
+This module contains python api to programmatically access exasol bucketfs service(s).
+
+
+.. attention:
+
+    If no python api is required, one can also use CLI tools like CURL and HTTPIE to access bucketfs services.
+
+    Example's using CURL and HTTPIE
+    -------------------------------
+
+    #. Listing buckets of a bucketfs service
+
+        HTTPIE:
+          $ http GET http://127.0.0.1:6666/
+
+        CURL:
+          $ curl -i http://127.0.0.1:6666/
+
+
+    #. List all files in a bucket
+    
+        HTTPIE:
+          $  http --auth w:write --auth-type basic GET http://127.0.0.1:6666/default
+
+        CURL:
+          $ curl -i -u "w:write" http://127.0.0.1:6666/default
+"""
+import warnings
+from collections import defaultdict
+from typing import Iterable, Mapping, MutableMapping
+from urllib.parse import urlparse
+
+from exasol_bucketfs_utils_python import BucketFsDeprecationWarning
+from exasol_bucketfs_utils_python.bucket_config import BucketConfig
+from exasol_bucketfs_utils_python.bucketfs_config import BucketFSConfig
+from exasol_bucketfs_utils_python.bucketfs_connection_config import (
+    BucketFSConnectionConfig,
+)
+from exasol_bucketfs_utils_python.buckets import list_buckets
+from exasol_bucketfs_utils_python.list_files import list_files_in_bucketfs
+
+
+class Service:
+    """Provides a simple to use api to access a bucketfs service.
+
+    Attributes:
+        buckets: lists all available buckets.
+    """
+
+    def __init__(self, url: str, credentials: Mapping[str, Mapping[str, str]] = None):
```

Comment:
Please introduce a credentials' provider abstraction here, because I have seen in some project, that these objects might get serialized from a UDF and then again deserialized, In that case, we should not store the credentials inside the object, but request it in the UDF via the get_connection when needed.